### PR TITLE
Validation

### PR DESCRIPTION
**Pull request by barakadanny**
In this PR, The following are the things achieved :
- Use validations for models.
- Write unit tests for models.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    it 'should not accept negative post_counter' do
+      subject.post_counter = -1
+      expect(subject).to_not be_valid
+    end
+
+    it 'should not accept non numerical post_counter' do
+      subject.post_counter = 'count'
+      expect(subject).to_not be_valid
+    end
```
Comment (omar-muhamad) [inline spec/models/user_spec.rb:41]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.), Please add a unit test for the `recent_three_posts` method in user model.

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    it 'should not accept non numerical likes_counter' do
+      subject.likes_counter = 'likes'
+      expect(subject).to_not be_valid
+    end
+
+    it 'should not accept non numerical comments_counter' do
+      subject.comments_counter = 'comments'
+      expect(subject).to_not be_valid
+    end
```
Comment (omar-muhamad) [inline spec/models/post_spec.rb:71]: - [ ] Same here, As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.), Please add a unit test for the `recent_five_posts` method in the post model.

File: README.md [hunk 3]
```diff
@@ -1,24 +1,48 @@
...
 
-* Deployment instructions
 
-* ...
+
+## Authors
+
+👤 **Baraka Danny (Me)**
+
+- GitHub: [@barakadan](https://github.com/barakadanny)
```
Comment (omar-muhamad) [inline README.md]: - [ ] Please use the new [Microverse README template](https://github.com/microverseinc/readme-template) instead of the old one.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @barakadanny,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: README.md (hunk 3 above)
Comment (barakadanny) [reply README.md]: @Omar-Muhamad 
Thank you for the review,
The changes have been implemented.

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @barakadanny,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

thumbs-up-agree


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**